### PR TITLE
Add RAOricutron

### PR DESCRIPTION
**Pull request by rzumer**
Opened as a new PR, because it is not possible to reopen old ones following a force push.

This is based on the master version of Oricutron with some modifications so it's simple to build with VS2019.

Again, hashing is full file, memory mapping is simply all the contiguous memory available, variable per model.

To-do:
- [ ] Switch to the repository-local `RAIntegration` (requires https://github.com/RetroAchievements/RAInterface/pull/1 and https://github.com/RetroAchievements/RAIntegration/pull/467)
- [ ] Adjust hotswapping protection to allow ejecting tapes and swapping to save tapes without reinserting the program (this might be needed for some titles, but can be left for later since it is a less restrictive model)
- [ ] Update Game Identification wiki page
- [ ] Update Systems Memory Maps wiki page
- [ ] Web-side updates (emulator entry for bug reports, etc.)

caricatur sucessfully developed a set for it, no blocking outstanding issues remain as far as I know, but I only tested tapes.

### CODE REVIEW

**Comment by rzumer** [comment]
I meant to make this branch point to my fork of RAIntegration and RAInterface but can't figure out how to actually get RAInterface changes merged into RAIntegration, it doesn't seem to work like a normal submodule, maybe I did something wrong.

**Comment by Jamiras** [comment]
RAInterface is a new submodule shared between RAIntegration and the emulators so they don't have to pull in the entire RAIntegration submodule. You should be the RAInterface submodule, and only the RAInterface submodule. Also, please consider creating a separate repository [as outlined here](https://github.com/RetroAchievements/RAInterface/wiki/Creating-a-new-emulator). It will be much easier to review the RA-specific changes and easier to update from source when necessary.

File: RAOricutron/.gitmodules [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[submodule "RAIntegration"]
+	path = RAIntegration
+	url = https://github.com/rzumer/RAIntegration.git
+	branch = oric
```
Comment (Jamiras) [inline RAOricutron/.gitmodules]: 1) You should be using the RAInterface submodule
2) You should be using the top-level submodule that's available to all emulators.

File: RAOricutron/RAIntegration/.clang-format [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿DisableFormat: false
```
Comment (Jamiras) [inline RAOricutron/RAIntegration/.clang-format]: it appears that your nested submodule was committed as raw code.

Comment (rzumer) [reply RAOricutron/RAIntegration/.clang-format]: Wow, that's an old mistake.

**Comment by rzumer** [comment]
Repository moved to https://github.com/RetroAchievements/oricutron and comments addressed in the `cheevos` branch.

**Pull request closed without merging**